helix-term/src/commands/syntax.rs

```diff
@@ -0,0 +1,478 @@
+use std::{
+    collections::HashSet,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use arc_swap::ArcSwapAny;
+use dashmap::DashMap;
```

Comment:
dashmap can be very slow to free memory (though it's fast on all other operations).

In my personal experience, [`scc`](https://docs.rs/scc) is better at memory reclamation, which is probably good for a picker (avoid lingering effects from mapping the whole workspace for example).

Of course, we probably want benchmarking results, it may not be an issue at all in practice

Reply:
For now I like dashmap just because it's already a transitive dependency. It'd be good to check if it's being wasteful in terms of memory and switch to something else. It's already a little bit overkill for what it's used for here